### PR TITLE
fixes label selector for deployment/pod relationship

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/controller/v16/resource/deployment/worker_deployment.go [hunk 1]
```diff
@@ -55,6 +55,13 @@ func newWorkerDeployments(customObject v1alpha1.KVMConfig) ([]*extensionsv1.Depl
 				},
 			},
 			Spec: extensionsv1.DeploymentSpec{
+				Selector: apismetav1.LabelSelector{
+					MatchLabels: map[string]string{
+						key.LabelApp: key.WorkerID,
+						"cluster":    key.ClusterID(customObject),
+						"node":       workerNode.ID,
```
Comment (kopiczko) [inline service/controller/v16/resource/deployment/worker_deployment.go]: Do we need to use "old-style" labels here?

**Review by kopiczko — APPROVED** [review]

File: service/controller/v16/resource/deployment/worker_deployment.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/v16/resource/deployment/worker_deployment.go]: That is the thing. We have to for the migration and to just make it work for now. When all pods have the new and the old labels, we can switch and remove the old ones completely.

**Review by calvix — APPROVED** [review]
lgtmiwfy

**Review by tuommaki — APPROVED** [review]
Tested this on `geckon`. This fix is part of a bigger puzzle. So works and is needed. :+1:

**Pull request merged**